addons/crm/tests/test_crm_lead_assignment.py

```diff
@@ -459,6 +459,35 @@ def test_assign_quota(self):
             "Assignment quota: no compensation and no limit anymore (do as asked even with 30 already assigned)"
         )
 
+    def test_random_choice(self):
+        """ Test no leads being assigned to any team member if weights list sums to 0"""
+
+        leads = self._create_leads_batch(
+            lead_type='lead',
+            user_ids=[False],
+            count=1
+        )
+
+        sales_team_4 = self.env['crm.team'].create({
+            'name': 'Sales Team 4',
+            'sequence': 15,
+            'use_leads': True,
+            })
+        sales_team_4_m1 = self.env['crm.team.member'].create({
+            'user_id': self.user_sales_salesman.id,
+            'crm_team_id': sales_team_4.id,
+            'assignment_max': 30,
+        })
+
+        sales_team_4_m1.lead_month_count = 50
+        leads.team_id = sales_team_4.id
```

Comment:
I need to hardcode the lead count for the team member here because the field is otherwise computed to 0 as per https://github.com/odoo/odoo/blob/047512e89163eabc03c3f144c9c46cf0292a7a55/addons/crm/models/crm_team_member.py#L31 (Because we're creating a brand new member for this test case)

For the lead, I'm using the helper method `_create_leads_batch` to create one lead with randomized data, but it does not have an arg that I can set to attach it to the particular sales team, which is why I need to set it here such that I'm targeting this sales team when running `_asign_and_convert_leads`.